openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -205,7 +205,8 @@ def celebrations(self):
         Returns a list of celebrations that should be performed.
         """
         return {
-            'first_section': CourseEnrollmentCelebration.should_celebrate_first_section(self.enrollment_object),
+            'first_section': (CourseEnrollmentCelebration.should_celebrate_first_section(self.enrollment_object)
+                              if courseware_mfe_first_celebration_is_active(self.course_key) else False),
```

Comment:
IIRC, I thought there was other stuff we needed to handle with the flag being off? Like we currently have a signal that creates this object and initializes it. Is this just saying, okay, keep the object, but we just don't return it until it's on? Not sure that's the best... if we have the flag turned off for a course, the database would still contain rows for all enrollments in that course despite it never actually being active. Food for thought.

Reply:
Ah yeah, excellent point, I totally forget about that code that I wrote.  :P

Search for create_course_enrollment_celebration -- we don't want to create a celebration for the learner if this flag is off.

Reply:
We also talked in grooming about wanting to delete celebration rows if we flip this switch off...  (or honestly, the REDIRECT_TO_COURSEWARE_MICROFRONTEND flag which also gates this feature)

Maybe we could do that manually? But if it's easy to plug into the flip turning off...  I don't know how often we're planning on flipping these off.